bundles/org.eclipse.equinox.console/src/org/eclipse/equinox/console/commands/ExportStateCommand.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Christoph Läubrich and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *     Christoph Läubrich - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.equinox.console.commands;
+
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Dictionary;
+import java.util.Hashtable;
+
+import org.apache.felix.service.command.CommandProcessor;
+import org.apache.felix.service.command.CommandSession;
+import org.apache.felix.service.command.Descriptor;
+import org.eclipse.osgi.container.ModuleDatabase;
+import org.osgi.framework.BundleContext;
+
+public class ExportStateCommand {
+
+	private BundleContext context;
+
+	public ExportStateCommand(BundleContext context) {
+		this.context = context;
+	}
+
+	public void startService() {
+		Dictionary<String, Object> dict = new Hashtable<>();
+		dict.put(CommandProcessor.COMMAND_SCOPE, "export");
+		dict.put(CommandProcessor.COMMAND_FUNCTION, new String[] { "exportFrameworkState" });
+		context.registerService(ExportStateCommand.class, this, dict);
+	}
+
+	@Descriptor("Exports the current framework state without the wiring information")
+	public void exportFrameworkState(CommandSession session, String path) throws IOException {
+		exportFrameworkState(session, path, false);
+	}
+
+	@Descriptor("Exports the current framework state, optionally with the wiring information")
+	public void exportFrameworkState(CommandSession session, String path, boolean persitWirings) throws IOException {
+		ModuleDatabase database = context.getBundle(0).adapt(ModuleDatabase.class);
+		PrintStream console = session.getConsole();
+		if (database != null) {
+			File file = new File(path);
+			console.println("Exporting ModuleDatabase to " + file.getAbsolutePath() + "...");
+			try (DataOutputStream stream = new DataOutputStream(new FileOutputStream(file))) {
+				database.store(stream, false);
```

Comment:
Can you show how this would be used?  I'm unsure how this will help if you don't actually persist the problematic wiring state which you want to debug or test with

Reply:
It seems I missed to pass the `persitWirings` by accident here, will fix that.

Beside that, if one wants to test the (fresh) resolve of some bundles the wiring are not needed (e.g. I have a set of bundles that do not resolve but should be) so the one can still see the bundles and use them.

But probably it won't hurt much to always export the wiring so it should be true by default?